### PR TITLE
ytdownloader: init at 3.17.3

### LINKED ISSUE #293327: Package request: ytDownloader
**Project description**

A modern GUI video and audio downloader supporting [hundreds of sites](https://github.com/yt-dlp/yt-dlp/blob/master/supportedsites.md) (using yt-dlp)

**Metadata**

* homepage URL: https://ytdn.netlify.app/
* source URL: https://github.com/aandrew-me/ytDownloader
* license:  GPL-3.0
* platforms: (any?)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by chewblacka**
## Description of changes
ytDownloader - A modern GUI video and audio downloader (utilizes yt-dlp and ffmpeg on the backend)
https://github.com/aandrew-me/ytDownloader
Init at 3.17.3

Closes https://github.com/NixOS/nixpkgs/issues/293327

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by chewblacka** [comment]
@ofborg eval

**Comment by chewblacka** [comment]
> also the app prints `Auto update: true` which we don't want, this should be false

What happens if you switch the auto-update off manually via the menu? Does it work then? Many thanks.

**Comment by mkg20001** [comment]
> > also the app prints `Auto update: true` which we don't want, this should be false
> 
> What happens if you disable auto-updates manually via the menu? Does it work then? Many thanks.

Due to the error I can't access the menu. Only option I have is try again button.

**Comment by chewblacka** [comment]
> Due to the error I can't access the menu. Only option I have is try again button.

Ok basically it's not an auto-update issue (I tried turning auto-update on and the program still worked for me).

If the program can't find `yt-dlp` then it tries to download it from the internet. Weirdly your build can't find `yt-dlp` and it also can't access the internet. Something very strange is going on. Are you on linux x86? Many thanks!

**Comment by mkg20001** [comment]
> > Due to the error I can't access the menu. Only option I have is try again button.
> 
> Ok basically it's not an auto-update issue (I tried turning auto-update on and the program still worked for me).
> 
> If the program can't find `yt-dlp` then it tries to download it from the internet. Weirdly your build can't find `yt-dlp` and it also can't access the internet. Something very strange is going on. Are you on linux x86? Many thanks!

yes, x86_64-linux.

it tries to download yt-dlp into the location of the binary

```
[pid 605369] openat(AT_FDCWD, "/nix/store/m6h6ay8i92ijn1lwkr6a4h0h1d5nav54-python3.11-yt-dlp-2023.12.30/bin/yt-dlp", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = -1 EROFS (Read-only file system)
```
(strace -f result/bin/ytdownloader)

check the code where it checks whether or not yt-dlp is already downloaded, there may be more to patch

**Comment by mkg20001** [comment]
Ok so it seems the offending line is `const fullYtdlpBinIsPresent = !!localStorage.getItem("fullYtdlpBinPresent");`

Simply patch that to true and it will never try to download yt-dlp

**Comment by chewblacka** [comment]
> Ok so it seems the offending line is `const fullYtdlpBinIsPresent = !!localStorage.getItem("fullYtdlpBinPresent");`
> 
> Simply patch that to true and it will never try to download yt-dlp

Much appreciated! Have patched that variable to true. Am unable to test it as it doesn't try to update on my system anyway. Hopefully that stops any updates. Many thanks!

**Comment by mkg20001** [comment]
> > Ok so it seems the offending line is `const fullYtdlpBinIsPresent = !!localStorage.getItem("fullYtdlpBinPresent");`
> > Simply patch that to true and it will never try to download yt-dlp
> 
> Much appreciated! Have patched that variable to true. Am unable to test it as it doesn't try to update on my system anyway. Hopefully that stops any updates. Many thanks!

works now, thx

File: pkgs/by-name/yt/ytdownloader/package.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        --add-flags $out/lib/node_modules/ytdownloader/main.js
+
+    install -Dm444 assets/images/icon.png $out/share/pixmaps/ytdownloader.png
+    install -Dm444 "${desktopItem}/share/applications/"* -t $out/share/applications
+  '';
+
+  meta = {
+    description = "A modern GUI video and audio downloader";
+    homepage = "https://github.com/aandrew-me/ytDownloader";
+    license = lib.licenses.gpl3;
```
Comment (mkg20001) [inline pkgs/by-name/yt/ytdownloader/package.nix:61]:
```suggestion
    license = lib.licenses.gpl3Only;
```

**Review by mkg20001 — APPROVED** [review]

**Pull request merged**